src/bluesearch/entrypoint/database/add.py

```diff
@@ -62,6 +65,50 @@ def run(
     with open(path, "rb") as f:
         article = pickle.load(f)  # nosec
 
+    # Article table.
+
+    # TODO At the moment, no identifiers are extracted. This is a patch waiting for it.
+    article_id = generate_uid((article.title,))
+
+    article_mapping = {
+        "article_id": article_id,
+        "title": article.title,
+        "authors": ", ".join(article.authors),
+        "abstract": "\n".join(article.abstract),
+    }
+    article_keys = article_mapping.keys()
+    article_fields = ", ".join(article_keys)
+    article_binds = f":{', :'.join(article_keys)}"
+
+    with engine.connect() as con:
```

Comment:
Maybe we can consider using `engine.begin()` to make use of [transactions](https://docs.sqlalchemy.org/en/14/core/connections.html#using-transactions)? If I read the SQLAlchemy documentation correctly it's safer for when an exception is raised within the `with`-block.

Reply:
There's also another `engine.connect()` further below, I guess the same applies there too.

Reply:
Using `connect()` is not coming from this PR. It is coming from #412.

Maybe @jankrepl could give more context about why #412 didn't use `begin()`?

Reply:
@jankrepl do you think we should use `engine.begin()` instead of `engine.connect()`?

Reply:
As discussed: 2df13d5.